raw-ipa-lib/src/report.rs

```diff
@@ -0,0 +1,137 @@
+use crate::threshold::DecryptionKey as ThresholdDecryptionKey;
+use crate::threshold::{Ciphertext, RistrettoPoint};
+
+use std::collections::HashMap;
+use std::fmt;
+
+pub struct EncryptedMatchkeys {
+    match_keys: HashMap<String, Ciphertext>,
+}
+
+impl PartialEq for EncryptedMatchkeys {
+    fn eq(&self, other: &Self) -> bool {
+        any_matches(self.match_keys.values(), &other.match_keys.values())
+    }
+}
+
+impl EncryptedMatchkeys {
+    pub fn count_matches(&self, other: &Self) -> usize {
+        n_matches(self.match_keys.values(), &other.match_keys.values())
+    }
+
+    pub fn from_matchkeys(match_keys: HashMap<String, Ciphertext>) -> EncryptedMatchkeys {
+        EncryptedMatchkeys { match_keys }
+    }
+
+    pub fn threshold_decrypt(
+        &self,
+        matchkey_decrypt: &ThresholdDecryptionKey,
+    ) -> EncryptedMatchkeys {
+        let partially_decrypted_matchkeys = self
+            .match_keys
+            .iter()
+            .map(|(p, emk)| (p.to_string(), matchkey_decrypt.threshold_decrypt(*emk)))
+            .collect();
+        EncryptedMatchkeys::from_matchkeys(partially_decrypted_matchkeys)
+    }
+
+    pub fn decrypt(&self, matchkey_decrypt: &ThresholdDecryptionKey) -> DecryptedMatchkeys {
+        let decrypted_matchkeys = self
+            .match_keys
+            .iter()
+            .map(|(p, emk)| (p.to_string(), matchkey_decrypt.decrypt(*emk)))
+            .collect();
+        DecryptedMatchkeys::from_matchkeys(decrypted_matchkeys)
+    }
+}
+
+impl fmt::Debug for EncryptedMatchkeys {
```

Comment:
You should be able to just add `#[derive(Debug)]` to the struct declaration.  It's a little more noise, but less code to write :)